### DISCUSSION TITLE
[Feature] Add Storage template Preset which adds time and date to the filename itself.

### DISCUSSION THREAD

**Discussion by Gulpman** [immich-app/immich#10838] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be nice to have a preset in the list for Storage templates which adds the time and date of a picture right in the filename of it instead of only the path to it.
To kill two birds with one stone (hopefully correctly translated into English), the format used by Synology NAS Photo app could be added to the preset list. It is doing exactly that. 

To keep the structure of dates within the file system something like the following could be added to the presets:

```
{{y}}/{{y}}-{{MM}}/IMG_{{y}}{{MM}}{{dd}}_{{HH}}{{mm}}{{ss}}
```

Resulting in e.g. IMG_5003.jpg being shot on 2024-07-04 at 10:43:33 to be stored at
/librarypath/2024/2024-07/IMG_20240704_104333.jpg

Having at least one template in the preset list which shows how to use the date and time on the filename would be helpful. It does not have to be the Synology format - I used it as I have a Synology running here and thought it would be convenient for those having a Synology NAS running to migrate to immich. 

Best
Sascha

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by cybericius** [comment]
Wow, there are no supporters of the idea? I think this would be the basic approach as original filenames typically have the date and timestamp data. Now I want to change the storage template back from Year\Month\filename to Year\filename so I could browse the images in shooting order in an external viewer, but it's not possible anymore. Supporting the idea!

**Reply by mmomjian** [reply]
You can set the storage template to whatever you want. This is mine: `{{filetypefull}}/{{y}}/{{MM}}_{{{album}}}_{{MMM}}/{{y}}-{{MM}}-{{dd}}_T{{HH}}{{mm}}{{ss}}_{{{filename}}}__{{assetId}}`

Closing this request as, a PR to add a template would be fine, but it's not major enough to require tracking in an FR.
